Framework/PythonInterface/plugins/algorithms/CompareSampleLogs.py

```diff
@@ -0,0 +1,209 @@
+from mantid.api import PythonAlgorithm, AlgorithmFactory, WorkspaceGroup, MatrixWorkspace
+from mantid.kernel import Direction, StringListValidator, StringArrayProperty, StringArrayLengthValidator,\
+    FloatBoundedValidator
+import mantid.simpleapi as api
+
+
+class CompareSampleLogs(PythonAlgorithm):
+    """
+    Compares specified sample logs for a given list of workspaces
+    """
+
+    def __init__(self):
+        """
+        Init
+        """
+        PythonAlgorithm.__init__(self)
+
+    def category(self):
+        """
+        Returns category
+        """
+        return "PythonAlgorithms;Utility\\Workspaces"
+
+    def name(self):
+        """
+        Returns name
+        """
+        return "CompareSampleLogs"
+
+    def summary(self):
+        return "Compares specified sample logs for a given list of workspaces."
+
+    def PyInit(self):
+        validator = StringArrayLengthValidator()
+        validator.setLengthMin(1)   # one group may be given
+        self.declareProperty(StringArrayProperty(name="InputWorkspaces", direction=Direction.Input, validator=validator),
+                             doc="Comma separated list of workspaces or groups of workspaces.")
+
+        self.declareProperty(StringArrayProperty(name="SampleLogs", direction=Direction.Input, validator=validator),
+                             doc="Comma separated list of sample logs to compare.")
+        self.declareProperty("Tolerance", 1e-3, validator=FloatBoundedValidator(lower=1e-7, upper=1.0),
+                             doc="Tolerance for comparison of double values.")
+
+        actions = ['warning', 'error']
+        self.declareProperty("DoNotMatchAction", "warning", StringListValidator(actions),
+                             doc="Action to perform if sample logs do not match.")
+        return
```

Comment:
What would be the reasonable output? For a big list of workspaces with a big list of properties to compare message about not matching properties can be long. Should I may be return true if all properties match and false if not?